quipucords/scanner/openshift/entities.py

```diff
@@ -250,5 +250,58 @@ def _parse_resource_string(value) -> tuple[int, str, bool]:
         return digits, power_name, ends_with_i
 
 
+class ClusterOperator(OCPBaseEntity):
+    """OCP Cluster Operator."""
+
+    _kind = "cluster-operator"
+    name: str
+    version: str = None
+    created_at: datetime.datetime = None
+    updated_at: datetime.datetime = None
+
+    @classmethod
+    def from_raw_object(cls, raw_object):
+        """Instantiate ClusterOperator using its equivalent api object."""
+
+        def _get_version(raw_object):
+            for version in raw_object.status.versions:
+                if version.name == "operator":
+                    return version.version
+            return None
+
+        return ClusterOperator(
+            name=raw_object.metadata.name,
+            created_at=raw_object.metadata.creationTimestamp,
+            updated_at=raw_object.status.conditions[0].lastTransitionTime,
+            version=_get_version(raw_object),
+        )
+
+
+class LifecycleOperator(ClusterOperator):
+    """Operator managed by Operator Lifecycle Manager (OLM)."""
+
+    _kind = "olm-operator"
+    package: str = None
+    source: str = None
+    channel: str = None
+    namespace: str = None
+
+    @classmethod
+    def from_raw_object(cls, raw_object):
+        """Instantiate OLM operator using its equivalent api object."""
+        installed_version = raw_object.status.currentCSV
```

Comment:
I saw this behavior with all operators installed from OperatorHub, but didn't found a documentation supporting this idea.

Regardless, I'd maintain as so. Behavior is pretty consistent - if it's not that, than it will fail very loudly (like the failure with memory) and we can fix it.

Reply:
Seems to be: https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/building-your-csv.md#csv-metadata. so `Operator . semantic version number`